src/graph/generate-chart-data.ts

```diff
@@ -0,0 +1,136 @@
+import { ChartData, ChartDataSets } from "chart.js";
+import { IDataset } from "@concord-consortium/lara-interactive-api";
+
+// Make Chart.js types a bit more specific (remove ? from properties definitions), so it's easier to work with them.
```

Comment:
If all you're doing is making optional properties required, consider using [`Required<T>`](https://mariusschulz.com/blog/mapped-type-modifiers-in-typescript#removing-the-mapped-type-modifier).

Reply:
Unfortunately Required makes all the properties required. I need only a few.